File: android2_lesson01/src/java/java/com/example/ama/android2_lesson01/ui/MainActivity.java
package com.example.ama.android2_lesson01.ui;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.example.ama.android2_lesson01.R;
import com.example.ama.android2_lesson01.model.Note;
import com.example.ama.android2_lesson01.ui.details.DetailsNoteFragment;
import com.example.ama.android2_lesson01.widget.NotesWidgetProvider;

/**
 * Class of main activity
 */

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        if (savedInstanceState == null) {
            Launcher.runListOfNotesFragment(this, false);
        }
        if (getIntent().getExtras() != null) {
            Note note = getIntent().getExtras().getParcelable(DetailsNoteFragment.TARGET_NOTE);
            if (note != null) {
                Launcher.runDetailsNoteFragment(this, true, note);
            }
        }
    }

    @Override
    protected void onStop() {
        NotesWidgetProvider.updateWidgets(getApplicationContext());
        super.onStop();
    }
}

File: android2_lesson01/src/kotlin/java/com/example/ama/android2_lesson01/ui/MainActivity.kt
package com.example.ama.android2_lesson01.ui

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.example.ama.android2_lesson01.R
import com.example.ama.android2_lesson01.model.Note
import com.example.ama.android2_lesson01.ui.details.DetailsNoteFragment
import com.example.ama.android2_lesson01.widget.NotesWidgetProvider

/**
 * Class of main activity
 */

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        savedInstanceState ?: Launcher.runListOfNotesFragment(this)
        val note = intent.extras?.getParcelable(DetailsNoteFragment.TARGET_NOTE) as Note?
        if (note != null) {
            Launcher.runDetailsNoteFragment(this, true, note)
        }
    }

    override fun onStop() {
        NotesWidgetProvider.updateWidget(applicationContext)
        super.onStop()
    }
}
